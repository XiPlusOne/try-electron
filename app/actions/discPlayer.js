// @flow
export const DISC_START_SPIN = 'DISC_START_SPIN';
export const DISC_STOP_SPIN = 'DISC_STOP_SPIN';
export const ROD_ON = 'ROD_ON';
export const ROD_OFF = 'ROD_OFF';
export const CHANGE_DISC = 'CHANGE_DISC';
export const CHANGE_SOUNDTRACK = 'CHANGE_SOUNDTRACK';
export const CHANGE_TITLE = 'CHANGE_TITLE';
export const CHANGE_DURATION = 'CHANGE_DURATION';
export const CHANGE_CURRENT_TIME = 'CHANGE_CURRENT_TIME';

export function discStartSpin() {
  return {
    type: DISC_START_SPIN
  };
}

export function discStopSpin() {
  return {
    type: DISC_STOP_SPIN
  };
}

export function rodOn() {
  return {
    type: ROD_ON
  };
}

export function rodOff() {
  return {
    type: ROD_OFF
  };
}

export function changeDisc(discPic: string) {
  return {
    type: CHANGE_DISC,
    data: discPic
  };
}

export function changeSoundTrack(soundTrack: string) {
  return {
    type: CHANGE_SOUNDTRACK,
    data: soundTrack
  };
}

export function changeTitle(title: string) {
  return {
    type: CHANGE_TITLE,
    data: title
  };
}

export function changeDuration(duration: number) {
  return {
    type: CHANGE_DURATION,
    data: duration
  };
}

export function changeCurrentTime(currentTime: number) {
  return {
    type: CHANGE_CURRENT_TIME,
    data: currentTime
  };
}
