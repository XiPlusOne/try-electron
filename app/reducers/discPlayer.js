// @flow
import { combineReducers } from 'redux';
import {
  DISC_START_SPIN,
  DISC_STOP_SPIN,
  ROD_ON,
  ROD_OFF
} from '../actions/discPlayer';
import type { Action } from './types';

function isDiscSpinning(state: boolean = false, action: Action) {
  switch (action.type) {
    case DISC_START_SPIN:
      return true;
    case DISC_STOP_SPIN:
      return false;
    default:
      return state;
  }
}

function isRodOn(state: boolean = false, action: Action) {
  switch (action.type) {
    case ROD_ON:
      return true;
    case ROD_OFF:
      return false;
    default:
      return state;
  }
}

export default combineReducers({
  isDiscSpinning,
  isRodOn
});
