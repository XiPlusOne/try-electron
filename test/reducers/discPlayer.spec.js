import discPlayer from '../../app/reducers/discPlayer';
import {
  DISC_START_SPIN,
  DISC_STOP_SPIN,
  ROD_ON,
  ROD_OFF,
  CHANGE_DISC,
  CHANGE_SOUNDTRACK,
  CHANGE_TITLE
} from '../../app/actions/discPlayer';

describe('reducers', () => {
  describe('discPlayer', () => {
    it('should handle initial state', () => {
      expect(discPlayer(undefined, {})).toMatchSnapshot();
    });

    it('should handle DISC_START_SPIN', () => {
      expect(
        discPlayer({ isDiscSpinning: false }, { type: DISC_START_SPIN })
      ).toMatchSnapshot();
    });

    it('should handle DISC_STOP_SPIN', () => {
      expect(
        discPlayer({ isDiscSpinning: true }, { type: DISC_STOP_SPIN })
      ).toMatchSnapshot();
    });

    it('should handle ROD_ON', () => {
      expect(
        discPlayer({ isRodOn: false }, { type: ROD_ON })
      ).toMatchSnapshot();
    });

    it('should handle ROD_OFF', () => {
      expect(
        discPlayer({ isRodOn: true }, { type: ROD_OFF })
      ).toMatchSnapshot();
    });

    it('should handle CHANGE_DISC', () => {
      expect(
        discPlayer({ discPic: '' }, { type: CHANGE_DISC, data: '233' })
      ).toMatchSnapshot();
    });

    it('should handle CHANGE_SOUNDTRACK', () => {
      expect(
        discPlayer({ soundTrack: '' }, { type: CHANGE_SOUNDTRACK, data: '233' })
      ).toMatchSnapshot();
    });

    it('should handle CHANGE_TITLE', () => {
      expect(
        discPlayer({ title: '' }, { type: CHANGE_TITLE, data: '233' })
      ).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(
        discPlayer(
          { isDiscSpinning: false, isRodOn: false },
          { type: 'unknown' }
        )
      ).toMatchSnapshot();
    });
  });
});
