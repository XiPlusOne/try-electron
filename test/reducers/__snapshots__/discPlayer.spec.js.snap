// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers discPlayer should handle DISC_START_SPIN 1`] = `
Object {
  "isDiscSpinning": true,
}
`;

exports[`reducers discPlayer should handle DISC_STOP_SPIN 1`] = `
Object {
  "isDiscSpinning": false,
}
`;

exports[`reducers discPlayer should handle ROD_OFF 1`] = `
Object {
  "isRodOn": false,
}
`;

exports[`reducers discPlayer should handle ROD_ON 1`] = `
Object {
  "isRodOn": true,
}
`;

exports[`reducers discPlayer should handle initial state 1`] = `
Object {
  "isDiscSpinning": false,
  "isRodOn": false,
}
`;

exports[`reducers discPlayer should handle unknown action type 1`] = `
Object {
  "isDiscSpinning": false,
  "isRodOn": false,
}
`;
